packages/teleterm/src/ui/TopBar/Identity/Identity.tsx

```diff
@@ -1,84 +1,150 @@
-import React, { useCallback, useMemo, useRef, useState } from 'react';
+import React, {
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+  useImperativeHandle,
+} from 'react';
 import styled from 'styled-components';
 import { Box } from 'design';
 import Popover from 'design/Popover';
-import { useIdentity } from './useIdentity';
+import { IdentityRootCluster, useIdentity } from './useIdentity';
 import { IdentityList } from './IdentityList/IdentityList';
 import { IdentitySelector } from './IdentitySelector/IdentitySelector';
-import { useKeyboardShortcuts } from 'teleterm/ui/services/keyboardShortcuts';
+import {
+  useKeyboardShortcuts,
+  useKeyboardShortcutFormatters,
+} from 'teleterm/ui/services/keyboardShortcuts';
 import { EmptyIdentityList } from './EmptyIdentityList/EmptyIdentityList';
-import { getClusterName } from 'teleterm/ui/utils';
+import { getClusterName, getUserWithClusterName } from 'teleterm/ui/utils';
+import * as tshd from 'teleterm/services/tshd/types';
 
-export function Identity() {
-  const selectorRef = useRef<HTMLButtonElement>();
-  const [isPopoverOpened, setIsPopoverOpened] = useState(false);
+export function IdentityContainer() {
   const {
     activeRootCluster,
     rootClusters,
     changeRootCluster,
     logout,
     addCluster,
   } = useIdentity();
+  const { getLabelWithShortcut } = useKeyboardShortcutFormatters();
 
-  const togglePopover = useCallback(() => {
-    setIsPopoverOpened(wasOpened => !wasOpened);
-  }, [setIsPopoverOpened]);
-
-  function withClose<T extends (...args) => any>(
-    fn: T
-  ): (...args: Parameters<T>) => ReturnType<T> {
-    return (...args) => {
-      setIsPopoverOpened(false);
-      return fn(...args);
-    };
-  }
+  const presenterRef = useRef<IdentityHandler>();
 
   useKeyboardShortcuts(
     useMemo(
       () => ({
-        'toggle-identity': togglePopover,
+        'toggle-identity': presenterRef.current?.togglePopover,
       }),
-      [togglePopover]
+      [presenterRef.current?.togglePopover]
```

Comment:
I'm not sure what I should pass here, `presenterRef.current` or the function itself, but I opted for the function.